deps.windows/70-vpl.ps1

```diff
@@ -0,0 +1,87 @@
+param(
+    [string] $Name = 'vpl',
+    [string] $Version = 'v2023.3.0',
+    [string] $Uri = 'https://github.com/oneapi-src/oneVPL.git',
+    [string] $Hash = 'e12ace9761bb52786409e830f619916b86e87fc5'
+)
+
+function Setup {
+    Setup-Dependency -Uri $Uri -Hash $Hash -DestinationPath $Path
+}
+
+function Clean {
+    Set-Location $Path
+
+    if ( Test-Path "build_${Target}" ) {
+        Log-Information "Clean build directory (${Target})"
+        Remove-Item -Path "build_${Target}" -Recurse -Force
+    }
+}
+
+function Configure {
+    Log-Information "Configure (${Target})"
+    Set-Location $Path
+
+    $Options = @(
+        $CmakeOptions
+        '-DBUILD_DISPATCHER_ONLY:BOOL=ON'
+        '-DBUILD_EXAMPLES:BOOL=OFF'
+        '-DBUILD_DISPATCHER_ONEVPL_EXPERIMENTAL:BOOL=OFF'
+        '-DBUILD_SHARED_LIBS:BOOL=OFF'
+    )
+
+    Invoke-External cmake -S . -B "build_${Target}" @Options
+}
+
+function Build {
+    Log-Information "Build (${Target})"
+    Set-Location $Path
+
+    $Options = @(
+        '--build', "build_${Target}"
+        '--config', $Configuration
+    )
+
+    if ( $VerbosePreference -eq 'Continue' ) {
+        $Options += '--verbose'
+    }
+
+    Invoke-External cmake @Options
+}
+
+function Install {
+    Log-Information "Install (${Target})"
+    Set-Location $Path
+
+    $Params = @{
+        ErrorAction = "SilentlyContinue"
+        Path = @(
+            "$($ConfigData.OutputPath)/lib"
+            "$($ConfigData.OutputPath)/include"
+        )
+        ItemType = "Directory"
+        Force = $true
+    }
+	
+    New-Item @Params *> $null
+
+    $Items = @(
+        @{
+            Path = "api/vpl"
+            Destination = "$($ConfigData.OutputPath)/include"
+            Recurse = $true
+            ErrorAction = 'SilentlyContinue'
+        }
+        @{
+            Path = "build_${Target}/$Configuration/vpl.lib"
+            Destination = "$($ConfigData.OutputPath)/lib"
+            ErrorAction = 'SilentlyContinue'
+        }
+    )
+	
+    $Items | ForEach-Object {
+        $Item = $_
+        Log-Output ('{0} => {1}' -f ($Item.Path -join ", "), $Item.Destination)
```

Comment:
I'm confused by what `$Item.Path -join ", "` is supposed to achieve, given that it's always a relative path string and not an array (given that this is code wrapped in a `ForEach-Object` block).

Reply:
I was using the other deps.windows files as a reference and this was the syntax that they used. If you have a better/simpler way to execute this function, I am open to any suggestions!

Reply:
Checked the sources - it is indeed unnecessary in this specific case, but we also have cases where the `Path` element of the hashtable is indeed an array:

```
    $Params = @{
        ErrorAction = "SilentlyContinue"
        Path = @(
            "$($ConfigData.OutputPath)/bin"
            "$($ConfigData.OutputPath)/lib"
        )
        ItemType = "Directory"
        Force = $true
    }
```

It's fine to use this as a generic pattern then.